### PR TITLE
Fix usec/msec discrepancy in timeout calculation

### PR DESCRIPTION
**Pull request by macintux**
Addresses #1293 (RIAK-2317)

### CODE REVIEW

File: src/riak_kv_w1c_worker.erl [hunk 1]
```diff
@@ -395,8 +395,10 @@ non_neg(X) ->
     max(0, X).
 
 remaining_put_timeout(Start, Timeout) ->
-    non_neg(Timeout -
-                non_neg(trunc(timer:now_diff(os:timestamp(), Start)))).
+    MicroSecondsSinceStart =
+        non_neg(trunc(timer:now_diff(os:timestamp(), Start))),
+    %% Convert to milliseconds before calculating the new timeout
+    non_neg(Timeout - (MicroSecondsSinceStart div 1000)).
```
Comment (hazen) [inline src/riak_kv_w1c_worker.erl:401]: Timeout is in milliseconds?  Looks like it.

Comment (macintux) [inline src/riak_kv_w1c_worker.erl:401]: Yes, Erlang timeouts are in milliseconds.

Comment (hazen) [inline src/riak_kv_w1c_worker.erl:401]: But timestamps are in microseconds, hence the confusion. Got it.

**Comment by hazen** [comment]
:+1: de3da45

**Comment by hazen** [comment]
@borshop merge

**Pull request merged**